plugins/catalog-backend-module-github/src/providers/GitHubEntityProvider.ts

```diff
@@ -185,9 +185,12 @@ export class GitHubEntityProvider implements EntityProvider {
     const repositoryFilter = this.config.filters?.repository;
 
     const matchingRepositories = repositories.filter(r => {
-      return !r.isArchived && repositoryFilter
-        ? repositoryFilter?.test(r.name)
-        : true && r.defaultBranchRef?.name;
+      return (
+        !r.isArchived &&
+        (repositoryFilter
+          ? repositoryFilter?.test(r.name)
+          : true && r.defaultBranchRef?.name)
```

Comment:
Hm. Even after the fix, it's surprising to me that the old code checked for the presence of a main branch _only if_ there was no repository filter.

Reply:
I think this should be

```ts
!r.isArchived &&
(!repositoryFilter || repositoryFilter.test(r.name)) &&
r.defaultBranchRef?.name
```